jenkins/Jenkinsfile

```diff
@@ -0,0 +1,178 @@
+def install_test_dependencies() {
+  echo 'installing test dependencies'
+  sh """#!/bin/bash
+    ~/venv/bin/pip install -r dev-requirements.txt
+    ~/venv/bin/pip install -r test-requirements.txt
+    ~/venv/bin/pip install -e .
+  """
+}
+def pytest(){
+  echo 'running pytest'
+  sh '''
+    ~/venv/bin/pytest \
+      --cov-report term-missing \
+      --cov=cloudify_cli \
+      cloudify_cli/tests \
+      --junitxml=test-results/cloudify_cli.xml
+  '''
+}
+
+@Library('pipeline-shared-library') _
+pipeline {
+  agent {
+    kubernetes {
+      label 'cli'
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+
+  options {
+    checkoutToSubdirectory('cloudify-cli')
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+  
+  environment {
+    PATH = "/root/.local/bin:$PATH"
+    PROJECT = 'cloudify-cli'
+    WORKSPACE = "${env.WORKSPACE}"
+    PROJECT_DIR = "${env.WORKSPACE}/project"
+    BUILD_DIR = "/tmp/build"
+    RESULT_DIR = "/tmp/result"
+    CLOUDIFY_PACKAGE_RELEASE = "1"
+    CLOUDIFY_VERSION = getVersion("master").trim()
+  }
+
+  stages {
+    stage('flake8, build rpm and deb-build') {
+      parallel {
+        stage('flake8_py27') {
+          steps {
+            sh script: "mkdir -p ${env.WORKSPACE}/flake8 && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/flake8", label: "copying repo to seperate workspace"
+            container('py27') {
+              dir("${env.WORKSPACE}/flake8") {
+                echo 'Install and run flake8'
+                sh '''
+                  pip install flake8 --user
+                  flake8 cloudify_cli
+                '''
+              }
+            }
+          }
+        }
+        stage('flake8_py36') {
+          steps {
+            sh script: "mkdir -p ${env.WORKSPACE}/flake8 && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/flake8", label: "copying repo to seperate workspace"
+            container('py36') {
+              dir("${env.WORKSPACE}/flake8") {
+                echo 'Install and run flake8'
+                sh '''
+                  pip install flake8 --user
+                  flake8 cloudify_cli
+                '''
+              }
+            }
+          }
+        }
+        stage('build-rpm') {
+          steps {
+            container('rpmbuild') {
+              sh """
```

Comment:
why """, use ''' here - I see you don't need interpolation

Reply:
also better to combine with 'Installing setup tools' shell or write echo before this one

Reply:
Thanks,
Changed